homeassistant/components/philips_js/light.py

```diff
@@ -0,0 +1,411 @@
+    )
+
+
+def _get_settings(style: AmbilightCurrentConfiguration):
+    """Extract the color settings data from a style."""
+    if style["styleName"] in ("FOLLOW_COLOR", "Lounge light"):
+        return style["colorSettings"]
+    if style["styleName"] == "FOLLOW_AUDIO":
+        return style["audioSettings"]
+    return None
+
+
+def _parse_effect(effect: str):
+    style, _, algorithm = effect.partition(EFFECT_PARTITION)
+    if style == EFFECT_MODE:
+        return EFFECT_MODE, algorithm, None
+    algorithm, _, expert = algorithm.partition(EFFECT_PARTITION)
+    if expert:
+        return EFFECT_EXPERT, style, algorithm
+    return EFFECT_AUTO, style, algorithm
+
+
+def _get_effect(mode: str, style: str, algorithm: str | None):
+    if mode == EFFECT_MODE:
+        return f"{EFFECT_MODE}{EFFECT_PARTITION}{style}"
+    if mode == EFFECT_EXPERT:
+        return f"{style}{EFFECT_PARTITION}{algorithm}{EFFECT_PARTITION}{EFFECT_EXPERT}"
+    return f"{style}{EFFECT_PARTITION}{algorithm}"
+
+
+def _is_on(mode, style, powerstate):
+    if mode in (EFFECT_AUTO, EFFECT_EXPERT):
+        if style in ("FOLLOW_VIDEO", "FOLLOW_AUDIO"):
+            return powerstate in ("On", None)
+        if style == "OFF":
+            return False
+        return True
+
+    if mode == EFFECT_MODE:
+        if style == "internal":
+            return powerstate in ("On", None)
+        return True
+
+    return False
+
+
+def _is_valid(mode, style):
+    if mode == EFFECT_EXPERT:
+        return style in EFFECT_EXPERT_STYLES
+    return True
+
+
+def _average_pixels(data):
+    """Calculate an average color over all ambilight pixels."""
+    color_c = 0
+    color_r = 0.0
+    color_g = 0.0
+    color_b = 0.0
+    for layer in data.values():
+        for side in layer.values():
+            for pixel in side.values():
+                color_c += 1
+                color_r += pixel["r"]
+                color_g += pixel["g"]
+                color_b += pixel["b"]
+
+    if color_c:
+        color_r /= color_c
+        color_g /= color_c
+        color_b /= color_c
+        return color_r, color_g, color_b
+    return 0.0, 0.0, 0.0
+
+
+class PhilipsTVLightEntity(CoordinatorEntity, LightEntity):
+    """Representation of a Philips TV exposing the JointSpace API."""
+
+    def __init__(
+        self,
+        coordinator: PhilipsTVDataUpdateCoordinator,
+        system: dict[str, Any],
+        unique_id: str,
+    ):
+        """Initialize light."""
+        self._tv = coordinator.api
+        self._hs = None
+        self._brightness = None
+        self._system = system
+        self._unique_id = unique_id
+        self._coordinator = coordinator
+        super().__init__(coordinator)
+
+        self._update_from_coordinator()
+
+    @property
+    def name(self):
+        """Return the device name."""
+        return self._system["name"]
+
+    @property
+    def unique_id(self):
+        """Return unique identifier if known."""
+        return self._unique_id
```

Comment:
https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L171
https://github.com/home-assistant/core/blob/dev/homeassistant/components/light/__init__.py#L603-L616

```suggestion
        self._system = system
        self._attr_name = system["name"]
        self._attr_unique_id = unique_id
        self._coordinator = coordinator
        super().__init__(coordinator)

        self._update_from_coordinator()
```